app/src/main/java/com/okta/oidc/example/PlainActivity.java

```diff
@@ -391,13 +450,17 @@ protected void onResume() {
         if (mWebAuth.isInProgress()) {
             showNetworkProgress(true);
         }
+        if (mBiometric.isChecked() && !mCurrentEncryptionManager.isUserAuthenticatedOnDevice()) {
+            showKeyguard();
+        }
     }
 
     @Override
     protected void onStop() {
         super.onStop();
+        getSharedPreferences(PlainActivity.class.getName(), MODE_PRIVATE).edit()
```

Comment:
would be more reliable to save this right when it's clicked. If there are any crashes we'll never save the state changes.